### ISSUE TITLE
Exception in thread "main" java.lang.NoClassDefFoundError: io/confluent/connect/storage/StorageSinkConnectorConfig

### ISSUE THREAD

**Issue by airztz** [confluentinc/kafka-connect-storage-cloud#99]
I’m new to kafka connector and currently I want to set up a connection between s3 and my own kafka cluster(not using confluent distribute).

I try to install the confluent s3 connector in this way: 

Copy the folder confluent-3.3.0/share/java/kafka-connect-s3 to my plugin.path, which I specified in worker.properties

And then launch my connect worker in distributed mode, it shows the bug as below:

INFO Loading plugin from: /home/ec2-user/kafka-connect-plugins/kafka-connect-s3 (org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader:176)
**Exception in thread "main" java.lang.NoClassDefFoundError: io/confluent/connect/storage/StorageSinkConnectorConfig**
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at org.apache.kafka.connect.runtime.isolation.PluginClassLoader.loadClass(PluginClassLoader.java:54)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getConstructor0(Class.java:3075)
	at java.lang.Class.newInstance(Class.java:412)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.getPluginDesc(DelegatingClassLoader.java:245)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.scanPluginPath(DelegatingClassLoader.java:224)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.scanUrlsAndAddPlugins(DelegatingClassLoader.java:198)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.registerPlugin(DelegatingClassLoader.java:190)
	at org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader.initLoaders(DelegatingClassLoader.java:150)
	at org.apache.kafka.connect.runtime.isolation.Plugins.<init>(Plugins.java:47)
	at org.apache.kafka.connect.cli.ConnectDistributed.main(ConnectDistributed.java:63)
Caused by: java.lang.ClassNotFoundException: io.confluent.connect.storage.StorageSinkConnectorConfig
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at org.apache.kafka.connect.runtime.isolation.PluginClassLoader.loadClass(PluginClassLoader.java:62)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 22 more

Looks like the worker can found that plugin folder, but looks like that we missed some parts, am I misunderstanding the way to install a kafka connector?

**Comment by airztz** [comment]
Follow up, looks like all other plugins are fine except s3 connector

<img width="423" alt="screen shot 2017-09-28 at 12 01 51 am" src="https://user-images.githubusercontent.com/11896105/30984132-55ecb6ec-a441-11e7-90e0-45853a4d8951.png">
<img width="586" alt="screen shot 2017-09-28 at 12 00 46 am" src="https://user-images.githubusercontent.com/11896105/30984140-5a21cbb2-a441-11e7-8d14-f27285e24525.png">

**Comment by kkonstantine** [comment]
Hi @airztz, 

the S3 connector does not bundle `kafka-connect-storage-common` with its dependencies. This has the implication that, when using `plugin.path` with the S3 connector, you'd need to add a symlink to point to the equivalent `kafka-connect-storage-common` directory of jars (copying all the jars would also work obviously). 

This will be reflected in the next CP where `plugin.path` will be enabled by default. 

Feel free to close this issue if the above resolves the failures you are experiencing.

**Comment by airztz** [comment]
@kkonstantine 
Thanks for your reply. I noticed that.

I solved this issue already by this way:

Instead of using confluent package and copy and paste confluent-3.3.0/share/java/kafka-connect-s3 into plugin.path, I build the kafka-connect-s3 package myself locally, then copy and paste 
kafka-connect-storage-cloud-3.3.0/kafka-connect-s3/target/kafka-connect-s3-3.3.0-development/share/java/kafka-connect-s3 into plugin.path of worker.properties
<img width="446" alt="screen shot 2017-10-02 at 2 26 34 pm" src="https://user-images.githubusercontent.com/11896105/31100329-e2190d2a-a77d-11e7-9be5-43f8c4026c0c.png">

That covers everything kafka-connect-s3 would need to run at any cluster.

Confluent package contains kafka-connect-s3-3.3.0-package/share/java/kafka-connect-s3, which does not cover all its necessary dependencies so it fails to deploy.

plugin.path works well at Kafka 0.11.01
